tests/integration/nupic/algorithms/extensive_tm_test_base.py

```diff
@@ -188,22 +189,26 @@ class ExtensiveTemporalMemoryTest(AbstractTemporalMemoryTest):
   when presented with the second A and B is different from the representation
   in the first presentation. [TODO]
   """
+  __metaclass__ = ABCMeta
 
   VERBOSITY = 1
-  DEFAULT_TM_PARAMS = {
-    "columnDimensions": [100],
-    "cellsPerColumn": 1,
-    "initialPermanence": 0.8,
-    "connectedPermanence": 0.7,
-    "minThreshold": 11,
-    "maxNewSynapseCount": 11,
-    "permanenceIncrement": 0.4,
-    "permanenceDecrement": 0,
-    "activationThreshold": 11,
-    "seed": 42
-  }
-  PATTERN_MACHINE = PatternMachine(100, range(21, 26), num=300)
 
+  def getPatternMachine(self):
+    return PatternMachine(100, range(21, 26), num=300)
+
+  def getDefaultTMParams(self):
+    return {
+      "columnDimensions": [100],
+      "cellsPerColumn": 1,
+      "initialPermanence": 0.8,
+      "connectedPermanence": 0.7,
+      "minThreshold": 11,
+      "maxNewSynapseCount": 11,
+      "permanenceIncrement": 0.4,
+      "permanenceDecrement": 0,
+      "activationThreshold": 11,
+      "seed": 42
```

Comment:
I guess personal preference but I like putting trailing comma on last item in list/dict like this so when you add or remove lines you don't have to maintain the last line differently.